src/Qir/Tests/QIR-dynamic/qsharp/qir-test-assert.qs

```diff
@@ -1,15 +1,363 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+// https://docs.microsoft.com/en-us/azure/quantum/concepts-the-qubit#single-qubit-operations    Qubit and Operations
+// https://docs.microsoft.com/en-us/azure/quantum/user-guide/libraries/standard/prelude         Operation Details, Multi-Qubit Operations
+
 namespace Microsoft.Quantum.Testing.QIR  {
 
     open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Intrinsic;
+
+    // To avoid the potentially circular dependency on QuantumLibraries repo we redefine the following {
+    // Identifiers from Microsoft.Quantum.Arrays:
+    // QuantumLibraries\Standard\src\Arrays\Arrays.qs:
+    function Head<'A> (array : 'A[]) : 'A {
```

Comment:
Since our QIR RT is going public, and the users are likely to use our tests as the examples of how to do certain things, I wanted to keep my test code as close as possible to an ordinary code calling Q# library, and to keep my **re**definitions of library calls as close as possible to the original definitions. Such that if one needs to understand what they are for and how they work, they could just open the corresponding documentation page (and compare my redefinitions with the original definitions, if they need to investigate the misbehavior).
For now I'd prefer to keep my *re*definitions as close as possible to the original definitions.

Reply:
Fair enough; we've generally not taken that strategy so far, though, so as to prevent duplicating too much code between the runtime and libraries repos. I'll defer to @bettinaheim on preferred strategies for unit testing in such contexts.

Reply:
Code duplications as part of tests are fine - though we should avoid any duplications outside tests. To keep our repo hierarchy as is (no dependency on the libraries repo in the runtime), we to some degree need to have similar constructs as they are used in the libraries duplicated as part of the tests.